### ISSUE TITLE
Error while i run training

### ISSUE THREAD

**Issue by hnnam0906** [kaist-silab/equity-transformer#1]
Hi @Leaveson, @hyeonahkimm, 

I get the following issue while running the training command in the readme
"python run.py --graph_size 50 --problem mtsp --run_name 'mtsp50' --agent_min 2 --agent_max 10"

assert not torch.isnan(log_p).any()
AssertionError
in _get_log_p method

![image](https://github.com/user-attachments/assets/01c3e074-e950-40fe-a8bc-cd169a9fdec6)

As far as i know all the log_p are NaNs so it throws the error. Please kindly help to investigate and correct the issue. 
Thanks

**Comment by hnnam0906** [comment]
Hi @Leaveson , @hyeonahkimm,

I hope you are doing well.
Your solution is really nice and i would like to use it. Can you kindly investigate and correct the above issue so that i can train it by myself?

Thanks

**Comment by hnnam0906** [comment]
Hi @Leaveson, @hyeonahkimm,

Could you help to solve the above issue?
I look forward to hearing from you.

Thanks.

**Comment by hyeonahkimm** [comment]
Hi @hnnam0906 , 

When I trained the model with the same script, the error didn't occur until after finishing one epoch. However, I suspect the error might be due to numerical instability in the log softmax operation. When log p values are too large, the operation can give nan values. Please try adding clipping before the log softmax function. Let me know if you still experience the same problem.

Best,
